File: 2.py
f = 1
g = 1
s = 0
while True:
	tmp = g
	g = tmp + f
	f = tmp
	if f > 4000000:
		break;
	#print f
	if f % 2 == 0:
		s += f
print "sum : " + str(s)

File: 2.hs
fib n = fibhelper 1 1 n
    where fibhelper a b n
              | b < n =
                  let c = a + b
                  in [c] ++ (fibhelper b c (n-1))
              | otherwise = []

evenfibsum n = sum (filter even (fib n))

maxfib = 4000000

main = do
	putStrLn (show (evenfibsum maxfib))
